hbase-client/src/main/java/org/apache/hadoop/hbase/client/RawAsyncHBaseAdmin.java

```diff
@@ -3505,6 +3506,36 @@ public <S, R> CompletableFuture<R> coprocessorService(Function<RpcChannel, S> st
     return future;
   }
 
+  @Override
+  public <S, R> CompletableFuture<Map<ServerName, Object>> coprocessorServiceOnAllRegionServers(
+    Function<RpcChannel, S> stubMaker, ServiceCaller<S, R> callable) {
+    CompletableFuture<Map<ServerName, Object>> future = new CompletableFuture<>();
+    FutureUtils.addListener(getRegionServers(), (rses, e1) -> {
+      if (e1 != null) {
+        future.completeExceptionally(e1);
+        return;
+      }
+      Map<ServerName, Object> resultMap = new HashMap<>();
```

Comment:
could use `ConcurrentHashMap` as well, so we don't have to synchronize the whole map

Reply:
@Apache9 do you have any concern with this? just thought of checking with you since you synchronized whole map in your sample.